Weather homepage

R4Ajeti opened the pull request:
Create a weather forecast site using the weather API 'https://openweathermap.org' from the previous lesson. You should be able to search for a specific location and toggle displaying the data in Fahrenheit or Celsius.

R4Ajeti committed:
INITIALIZED: React app
Use create-react-app to initialize a new React app
Documentation: https://github.com/facebook/create-react-app
Delete any unnecessary file or asset

R4Ajeti committed:
ADDED: searchWeather and init functions
1. Set up a blank HTML document with the appropriate links to your JavaScript and CSS files.
2. Write the functions that hit the API. You’re going to want functions that can take a location or zip code and return the weather data for that location. For now, just console.log() the information.
3. Write the functions that process the JSON data you’re getting from the API and return an object with only the data you require for your app.
4. Set up a simple form that will let users input their location and will fetch the weather info (still just console.log() it).
5. Display the information on your webpage!
6. Add any styling you like!

R4Ajeti committed:
ADDED: Finding weather by coords
Finding weather by geographic coordinates latitude and longitude

shloch reviewed (changes requested):
# hello @R4Ajeti 

I can see you are already a react user. This project is well executed. Good use of asynchronous js

![Screen Shot 2020-01-27 at 12 50 15 AM](https://user-images.githubusercontent.com/40712459/73143865-48962a80-409f-11ea-93be-a3381c9f958d.png)

Great job 

## [required changes]

- which unit of measurement is on display on screen, is it degree CELCIUS or FAHREINHEIT ?? (specify)

- there should be a toggle button to see results in either CELCIUS or FAHREINHEIT ( [see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png)) 

- Configure Stickler into this repo ...Since you are using react, [use this](https://github.com/microverseinc/linters-config/tree/master/react-redux)
- Please update your README explaining how to  to have project settup locally 😊

## Waiting for another re-review,

Happy coding

R4Ajeti committed:
ADDED: Linter/Stickler configuration
Stickler & linter throw:
https://github.com/microverseinc/linters-config/tree/master/react-redux

R4Ajeti committed:
ADDED: Toggle => Celcius/Ferinheight
    celciusToFarenheit() - function converts Celcius to Ferinheight from its formula
Accepts 1 or 2 arguments:
@args[0] holds value to convert,
@args[1] holds if is given as false or -1 does the inreverse ( converts from Ferinheight to Celcius)

    toggleFunc changes the states depending on:
@lastTemp - lastest value of temperature captured,
@tempType - determines if value is as Celcius or Ferinheight

R4Ajeti committed:
UPDATED: README.md

bellom commented on .eslintrc:
```diff
@@ -0,0 +1,6 @@
+{
```
You should remove this file so far you're not using it, you're using `eslintrc.json` in `stickler.yml`

bellom reviewed (changes requested):
## Hi @R4Ajeti  :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

**I quote from the previous reviewer:**
```
there should be a toggle button to see results in either CELCIUS or FAHRENHEIT :

Configure Stickler into this repo ...Since you are using react, use this 

Please update your README explaining how to have the project set up locally 😊 ✅ 
```

**Kindly work on the following;**
1. There is a bug using the toggle button.
Clicking the toggle button changes the value of the original result on second search.
Note: First search is working fine, but once I search for another city and click the toggle button it will change the result back to the first search value.

2. You need to fix the linter's error.
![image](https://user-images.githubusercontent.com/31897434/73417568-ca44bd00-4318-11ea-8a5e-fedabd629b8c.png)
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/

**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

R4Ajeti committed:
FIXING: Linter errors
Re-installed npm, then installed eslint throw:
https://github.com/microverseinc/linters-config/tree/master/react-redux#troubleshooting

R4Ajeti committed:
FIXING: Linter errors

R4Ajeti committed:
FIXED: Linter

R4Ajeti committed:
FIXED: Linter errors

R4Ajeti committed:
FIXED: Second search bug, ADDED: AlertMessage
Bootstrap, JQuery and SASS installed
components re-structured

R4Ajeti committed:
FIXING: Fixing linter errors - 'rules off'

R4Ajeti committed:
FIXED: Linter errors

bellom commented:
@R4Ajeti kindly reply your messages on slack

R4Ajeti committed:
FIXED-TSE: Linter errors

bellom commented:
Have you tried the commands I sent?
`npx eslint --fix`

bellom reviewed (approved):
## Hi @R4Ajeti  :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next project :slightly_smiling_face:_ 

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**
